pages/DynamicPOC/index.tsx

```diff
@@ -76,71 +77,85 @@ const DynamicPOC: React.FC = () => {
-        ? _getInputAnswerId(event.target.name)
-        : event.target.value,
+      answerId: isUserInput ? '0' : event.target.value,
       userAnswer: isUserInput ? event.target.value : undefined,
     }
 
     setCurrentAnswer(answer)
   }
 
-  function _getInputAnswerId(questionId: string): string {
-    const question: Question = questions[
-      questionId as QuestionsKeys
-    ] as Question
-    return question.answers[0].toString() // TODO: Probably error check this or something
-  }
-
   /**
    * handles getting the next question based on current question's answer
    */
   function _handleNext() {
-    if (formValues.formAnswers.hasOwnProperty(currentAnswer.questionId)) {
+    if (
+      !currentAnswer ||
+      currentAnswer.questionId !== currentQuestion.id.toString()
+    ) {
+      return
+    }
+    if (formValues.formAnswers.hasOwnProperty(currentQuestion.id)) {
       _handleQuestionExists()
     } else {
+      const nextQuestion = getNextQuestion(
+        questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
+      )
       setQuestionHistory((questionHistory) => [
         ...questionHistory,
-        currentQuestion,
+        nextQuestion,
       ])
-      setCurrentQuestion(getNextQuestion(currentAnswer.answerId as AnswersKeys))
-    }
-    formValues.formAnswers[currentAnswer.questionId] = currentAnswer
-    if (formValues.formAnswers.hasOwnProperty(currentQuestion.id)) {
-      setCurrentIndex(currentIndex + 1)
+      formValues.formAnswers[currentQuestion.id] = currentAnswer
+      setCurrentQuestion(nextQuestion)
+      if (formValues.formAnswers[nextQuestion.id]) {
+        setCurrentAnswer(formValues.formAnswers[nextQuestion.id])
+      }
     }
+    // if (formValues.formAnswers.hasOwnProperty(currentQuestion.id)) {
+    setCurrentIndex(currentIndex + 1)
+    // }
   }
 
   /**
    * Modifies question history and routes form depending on whether answer has been changed
    */
   function _handleQuestionExists() {
     if (
-      formValues.formAnswers[currentAnswer.questionId]['answerId'] !==
+      formValues.formAnswers[currentQuestion.id]['answerId'] !==
       currentAnswer['answerId']
     ) {
       for (let i = currentIndex + 1; i < questionHistory.length; i++) {
         delete formValues.formAnswers[questionHistory[i].id]
       }
       const questionSlice = questionHistory.slice(0, currentIndex + 1)
-      setQuestionHistory([...questionSlice, currentQuestion])
-      setCurrentQuestion(getNextQuestion(currentAnswer.answerId as AnswersKeys))
+      const nextQuestion = getNextQuestion(
+        questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
+      )
+      setQuestionHistory([...questionSlice, nextQuestion])
+      formValues.formAnswers[currentQuestion.id] = currentAnswer
+      setCurrentQuestion(nextQuestion)
+      if (formValues.formAnswers[nextQuestion.id]) {
+        setCurrentAnswer(formValues.formAnswers[nextQuestion.id])
+      }
     } else {
```

Comment:
Maybe this is just because of the PR readability, but is the else statement just empty here?

Reply:
I think it's the PR readability